sentence_transformers/SentenceTransformer.py

```diff
@@ -572,11 +578,48 @@ def get_sentence_features(self, *features):
         return self._first_module().get_sentence_features(*features)
 
     def get_sentence_embedding_dimension(self):
+        """
+        :return: The number of dimensions in the output of `encode`. If it's not known, it's `None`.
+        """
+        output_dim = None
         for mod in reversed(self._modules.values()):
             sent_embedding_dim_method = getattr(mod, "get_sentence_embedding_dimension", None)
             if callable(sent_embedding_dim_method):
-                return sent_embedding_dim_method()
-        return None
+                output_dim = sent_embedding_dim_method()
+                break
+        if self.output_dim is not None:
+            # The user requested truncation. If they set it to a dim greater than output_dim,
+            # no truncation will actually happen. So return output_dim insead of self.output_dim
+            return min(output_dim or np.inf, self.output_dim)
+        return output_dim
+
+    @contextmanager
+    def truncate_sentence_embeddings(self, output_dim: Optional[int]):
```

Comment:
This is quite clean, though I'm not sure if people will immediately draw the connection between `truncate_sentence_embeddings` and `output_dim`. Perhaps this informs us that e.g. `truncate_dim` is a stronger name than `output_dim`?

Reply:
Ah right. Let's go with `truncate_dim`

Reply:
Apologies, I should have been clearer, I think `truncate_sentence_embeddings` is fairly clear already, but perhaps we should rename `output_dim` to `truncate_dim`.

So,
```python
model = SentenceTransformer(
    "nomic-ai/nomic-embed-text-v1.5",
    trust_remote_code=True,
    truncate_dim=matryoshka_dim,
)
```

I'm curious about your thoughts here, too!